Nodejs/Product/Nodejs/SharedProject/ProjectNode.cs

```diff
@@ -2614,20 +2614,21 @@ protected virtual MSBuildResult InvokeMsBuild(string target)
             this.Site.GetUIThread().MustBeCalledFromUIThread();
 
             var result = MSBuildResult.Failed;
-            const bool designTime = true;
 
-            BuildSubmission submission = null;
-
-            try
+            // Do the actual Build
+            if (this.buildProject != null)
             {
-                // Do the actual Build
-                if (this.buildProject != null)
+                const bool designTime = true;
+                const bool requiresUIThread = true;
+
+                if (!TryBeginBuild(designTime, requiresUIThread))
```

Comment:
Either `designTime` or `requiresUIThread` is not required. The two places where `TryBeginBuild` is called, have the same value for both variables. We can remove one and refactor the method to make it a little bit more readable. I suggest removing `designTime`, as `requiresUIThread` is easy to understand its purpose.

Reply:
I'd prefer to retain `designTime` since one could reasonably conclude that `designTime` requires the UI thread, but I don't see how would deduce that `requiresUIThread` implies a design time build.

Reply:
I'll do this in a new PR.